Python/ceval_gil.c

```diff
@@ -625,10 +632,12 @@ _PyEval_AcquireLock(PyThreadState *tstate)
 }
 
 void
-_PyEval_ReleaseLock(PyThreadState *tstate)
+_PyEval_ReleaseLock(PyInterpreterState *interp, PyThreadState *tstate)
```

Comment:
The other key part of this change is here, where we pass in the interpreter separately from the thread state, which allows `tstate` to be NULL.